frontend/src/scenes/data-warehouse/ViewLinkModal.tsx

```diff
@@ -1,102 +1,146 @@
-                        <Field name="saved_query_id">
+                        <span className="l4">Joining Table</span>
+                        <Field name="joining_table_name">
                             <LemonSelect
                                 fullWidth
-                                options={viewOptions}
-                                onSelect={selectView}
-                                placeholder="Select a view"
+                                options={tableOptions}
+                                onSelect={selectJoiningTable}
+                                placeholder="Select a table"
                             />
                         </Field>
                     </div>
                 </div>
                 <div className="mt-3 flex flex-row justify-between items-center w-full">
                     <div className="w-50">
-                        <span className="l4">Table Key</span>
-                        <Field name="from_join_key">
-                            <LemonSelect fullWidth options={fromJoinKeyOptions} placeholder="Select a key" />
+                        <span className="l4">Source Table Key</span>
+                        <Field name="source_table_key">
+                            <LemonSelect
+                                fullWidth
+                                disabledReason={selectedSourceTableName ? '' : 'Select a table to choose join key'}
+                                options={sourceTableKeys}
+                                placeholder="Select a key"
+                            />
                         </Field>
                     </div>
                     <div className="mt-5">
                         <IconSwapHoriz />
                     </div>
                     <div className="w-50">
-                        <span className="l4">View Key</span>
-                        <Field name="to_join_key">
+                        <span className="l4">Joining Table Key</span>
+                        <Field name="joining_table_key">
                             <LemonSelect
                                 fullWidth
-                                disabledReason={selectedView ? '' : 'Select a view to choose join key'}
-                                options={toJoinKeyOptions}
+                                disabledReason={selectedJoiningTable ? '' : 'Select a table to choose join key'}
+                                options={joiningTableKeys}
                                 placeholder="Select a key"
                             />
                         </Field>
                     </div>
                 </div>
+                {sqlCodeSnippet && (
+                    <>
+                        <LemonDivider className="mt-4 mb-4" />
+                        <div className="mt-3 flex flex-row justify-between items-center w-full">
+                            <div className="w-full">
+                                <span className="l4">Field Name</span>
```

Comment:
for simplicity, maybe only show this if there's a conflict?

Reply:
It gets pre-populated, so 90% of the time it won't get changed I imagine, but I think it's nice that it's clear what the field will be

Reply:
Yeah, I agree with the goal, maybe we can hide this under some "Advanced" group of fields. There will be a case where the source table will already have a field that conflicts with the "default" field name we give the joining table, so having the ability to customize this is certainly important. 

We also need to figure out how to let users join on JSON objects, e.g. if I wanna join on `properties.some_field.user_id` - maybe just allowing hogql expressions works for this, but again, this will increase the complexity of the UI somewhat